### PR TITLE
chore: [ANDROSDK-1902] Remove remaining calendar usages 

### PR DESCRIPTION
**Pull request by marcamsn**
Solves [ANDROSDK-1902](https://dhis2.atlassian.net/browse/ANDROSDK-1902).

[ANDROSDK-1902]: https://dhis2.atlassian.net/browse/ANDROSDK-1902?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: core/src/main/java/org/hisp/dhis/android/core/event/internal/EventDateUtils.kt [hunk 1]
```diff
@@ -27,17 +27,20 @@
...
 import org.hisp.dhis.android.core.period.PeriodType
+import org.hisp.dhis.android.core.period.clock.internal.ClockProvider
 import org.hisp.dhis.android.core.period.internal.PeriodHelper
 import org.koin.core.annotation.Singleton
-import java.util.*
-import java.util.concurrent.TimeUnit
 
 @Singleton
-class EventDateUtils(
+internal class EventDateUtils(
```
Comment (vgarciabnz) [inline core/src/main/java/org/hisp/dhis/android/core/event/internal/EventDateUtils.kt:41]: Did you check this class wasn't used by the Android app?

File: core/src/main/java/org/hisp/dhis/android/core/arch/helpers/DateUtils.kt [hunk 2]
```diff
@@ -28,13 +28,17 @@
...
+import kotlinx.datetime.plus
 import kotlinx.datetime.toLocalDateTime
 import org.hisp.dhis.android.core.arch.dateformat.internal.SafeDateFormat
 import org.hisp.dhis.android.core.period.Period
 import org.hisp.dhis.android.core.period.PeriodType
-import org.hisp.dhis.android.core.period.internal.CalendarProviderFactory
-import java.util.*
+import java.util.Date
 
 object DateUtils {
```
[the remark this answers is not in the document]
Comment (marcamsn) [reply core/src/main/java/org/hisp/dhis/android/core/arch/helpers/DateUtils.kt:43]: Yes, I check it. :) 
It is used in a .kt class, and the only usage is the `DateUtils.SIMPLE_DATE_FORMAT`.

File: core/src/main/java/org/hisp/dhis/android/core/event/internal/EventDateUtils.kt (hunk 1 above)
Comment (marcamsn) [reply core/src/main/java/org/hisp/dhis/android/core/event/internal/EventDateUtils.kt:41]: Yes, there are no usages of this class.

File: core/src/main/java/org/hisp/dhis/android/core/arch/helpers/DateUtils.kt (hunk 2 above)
[the remark this answers is not in the document]
Comment (vgarciabnz) [reply core/src/main/java/org/hisp/dhis/android/core/arch/helpers/DateUtils.kt:43]: So, what do we do with this class? It is a class that was public and wasn't in an `.internal.` package, I think it was intentionally public. If we do this change, we have to document it as a breaking change in the release notes. Probably nobody is using this class, we could change it, but it is still the SDK api.

**Review by vgarciabnz — APPROVED** [review]

**Pull request merged**